src/components/SEO/MetaTags.jsx

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import PropTypes from "prop-types"
+import Helmet from "react-helmet"
+
+import Keywords from "./Keywords"
+import OpenGraph from "./OpenGraph"
+import Twitter from "./Twitter"
+
+const MetaTags = ({
+  description,
+  image,
+  lang,
+  keywords,
+  title,
+  twitter,
+  url,
+}) => (
+  <>
+    <Helmet htmlAttributes={{ lang }} title={title}>
+      <meta name="description" content={description} />
+      <meta name="title" content={title} />
+    </Helmet>
+    <Keywords keywords={keywords} />
```

Comment:
Why not just put everything into a single component, splitting this off is just creating entropy.

Reply:
Why do you say that?

Everything is currently in the same single SEO component, which is making things very hard to read and change IMO, because there's all this logic and tags floating around. In this specific case, by handing off the keywords we can make it optional (not render the tag) without having to pollute this level with branching in JSX.

IMO, this is actually reducing entropy by making things easier to read, understand, and more orderly (one component with the single responsibility of handling keywords). Nonetheless this is my opinion, would like to understand yours.

Reply:
It's ok. I just don't split components anymore this way, at all. Being easier to read is debatable. You have to open multiple files to understand what is going on. For logic in JSX you can just `condition && <Something/>` that for me is easier to read. I just do this because on the frontend you are always writing a bunch of stuff and if you keep with this pattern you might end up with thousands of components, where most of them are just used in one place.

I only split things off when they need to be reused on more than one place, or to isolate them for testing purposes. Or just have a single file with more than one component or render functions.

Reply:
I agree that's debatable. I don't consider the `condition && <Something />` easy to read, especially when it starts spawning multiple lines. I understand what you're saying about splitting components when they're to be used in more than one place, but in these cases you have a top-level component that's being reused, and you have nested components inside this one's directory that are meant to be used by it. Yes, you might get a lot more components, but the structure itself tells you how they're organized (the hierarchy).

As for the argument of a single one with multiple components/render functions, while I have nothing against it per se, I do refrain from doing it due to the length the files end up having. I would rather have hundreds of little files, with very easy to understand and self-contained components, then to have a single file spawning over 100 lines where you get lost easily. It's like taking the single-responsibility rule but to the file level, each file is in charge of one component so sub-components go to nested files.

Difference of opinion I reckon. :man_shrugging: